dev/domain-controller/FinalizeDeployment.ps1

```diff
@@ -20,3 +31,39 @@ Invoke-Command -Session $psSession -ScriptBlock {
     New-ADGroup -name $using:groupToJoin -GroupScope Global
 	Add-ADGroupMember -Identity $using:groupToJoin -Members $machineToJoin
 }
+
+
+# create self signed certificate
+$certLoc = 'cert:Localmachine\My'
+$cert = New-SelfSignedCertificate -certstorelocation $certLoc -dnsname local.teradici.com
+
+#generate pfx file
+$certPath = $certLoc + '\' + $cert.Thumbprint
+$certPfx = 'C:\WindowsAzure\mySelfSignedCert.pfx'
+$randomPswd = -join ((48..57) + (65..90) + (97..122) | Get-Random -Count 16 | % {[char]$_})
+$certPwd = ConvertTo-SecureString -String $randomPswd -AsPlainText -Force
+Export-PfxCertificate -Cert $certPath -FilePath $certPfx -Password $certPwd
+
+#read from pfx file and generate 64base encoded string
+$fileContentBytes = get-content $certPfx -Encoding Byte
+$fileContentEncoded = [System.Convert]::ToBase64String($fileContentBytes)
+
+# Login to azure
+$azurePwd = ConvertTo-SecureString $azurePassword -AsPlainText -Force
+$azureloginCred = New-Object -TypeName pscredential –ArgumentList $azureUserName, $azurePwd
+Login-AzureRmAccount -Credential $azureloginCred
+
+# deploy application gateway
+$parameters = @{}
+$parameters.Add(“subnetRef”, $subnetRef)
+$parameters.Add(“skuName”, "Standard_Small")
+$parameters.Add(“capacity”, 1)
+$parameters.Add(“backendIpAddressDefault”, "$backendIpAddressDefault")
+$parameters.Add(“backendIpAddressForPathRule1”, "$backendIpAddressForPathRule1")
+$parameters.Add(“pathMatch1”, "/pcoip-broker/*")
+$parameters.Add(“certData”, "$fileContentEncoded")
+$parameters.Add(“certPassword”, "$randomPswd")
+
+$deployName = -join ((65..90) + (97..122) | Get-Random -Count 12 | % {[char]$_})
```

Comment:
it normally work unless the parent deploy name is the same. I choose the random name to prevent the name conflict with parent name and this name does not show in portal